data/index.html

```diff
@@ -5,133 +5,125 @@
+        Last update: <span id="last_update"></span>
+      </p>
+    </div>
+  </div>
 
-<form>
-  Probe is
-  <select id="select_constraint">
-    <option value="is_in" selected="selected">recorded</option>
-    <option value="new_in">new</option>
-  </select>
-  in version
-  <select id="select_version">
-    <option id="version_any" value="any" selected="selected">any</option>
-  </select>
-  on channel
-  <select id="select_channel">
-    <option value="release" selected="selected">release</option>
-    <option value="beta">beta</option>
-    <option value="aurora">aurora</option>
-    <option value="nightly">nightly</option>
-  </select>
-  .
-  <br />
-  <input id="optout" type="checkbox" checked="checked">
-  <label for="optout">Show only opt-out measurements.</label>
-  <br />
-  Text search
-  <select id="search_constraint">
-    <option value="in_any" selected="selected">in any text field</option>
-    <option value="in_name">in name</option>
-    <option value="in_description">in description</option>
-  </select>
-  for:
-  <input id="text_search" name="text_search">
-</form>
+  <!-- Form -->
+  <div class="container-fluid">
+    <form class="ml-1 mt-3">
+      <div class="form-group form-inline">
+        Probe is
+        <select class="form-control ml-2 mr-2" id="select_constraint">
+          <option value="is_in" selected="selected">recorded</option>
+          <option value="new_in">new</option>
+        </select>
+        in version
+        <select class="form-control ml-2 mr-2" id="select_version">
+          <option id="version_any" value="any" selected="selected">any</option>
+        </select>
+        on channel
+        <select class="form-control ml-2 mr-2" id="select_channel">
+          <option value="release" selected="selected">release</option>
+          <option value="beta">beta</option>
+          <option value="aurora">aurora</option>
+          <option value="nightly">nightly</option>
+        </select>
+        .
+      </div>
+      <div class="form-group form-inline">
+        <input class="form-control mr-2" id="optout" type="checkbox" checked="checked">
+        <label for="optout">Show only opt-out measurements.</label>
+      </div>
+      <div class="form-group form-inline">
+        Text search
+        <select class="form-control ml-2 mr-2" id="search_constraint">
+          <option value="in_any" selected="selected">in any text field</option>
+          <option value="in_name">in name</option>
+          <option value="in_description">in description</option>
+        </select>
+        for:
+        <input class="form-control ml-2 mr-2" id="text_search" name="text_search">
+      </div>
+    </form>
+  </div>
 
-<div id="stats">
-</div>
+  <hr />
 
-<div id="measurements">
-</div>
+  <!-- Placeholders filled by JS -->
+  <div class="conainter-fluid">
+    <!-- I'm not sure what div tag is here for. -->
+    <div class="container ml-4" id="stats"></div>
```

Comment:
In `stats.js`, we pass this element to D3, which then draws the visualization into it.
I.e.: `var svg = d3.select("#stats");`

Reply:
Right. However, D3 is not pushing anything to the element on the index.html page, correct?

Reply:
Oh, i see!
Right, this seems unused.